### PR TITLE
[CT-804] Fix stateful order removal event ordering.

### PR DESCRIPTION
**Pull request by vincentwschau**
### Changelist
Add stateful order removals as block events rather than transaction events to fix their ordering in the produced indexer block message.

### Test Plan
Tested in staging upgrading from v4.1 to v5.x

### Author/Reviewer Checklist
- [ ] If this PR has changes that result in a different app state given the same prior state and transaction list, manually add the `state-breaking` label.
- [ ] If the PR has breaking postgres changes to the indexer add the `indexer-postgres-breaking` label.
- [ ] If this PR isn't state-breaking but has changes that modify behavior in `PrepareProposal` or `ProcessProposal`, manually add the label `proposal-breaking`.
- [ ] If this PR is one of many that implement a specific feature, manually label them all `feature:[feature-name]`.
- [ ] If you wish to for mergify-bot to automatically create a PR to backport your change to a release branch, manually add the label `backport/[branch-name]`.
- [ ] Manually add any of the following labels: `refactor`, `chore`, `bug`.

## Summary by CodeRabbit

- **Refactor**
	- Enhanced test cases for order placement, triggering, and removal with parameterized testing for various event scenarios.
- **Tests**
	- Improved reliability and coverage of testing for transaction and block event handling.
- **Documentation**
	- Updated internal documentation to align with revised testing strategies and event handling procedures.

### CODE REVIEW

**Review by teddyding — APPROVED** [review]

File: indexer/services/ender/src/lib/block-processor.ts [hunk 1]
```diff
@@ -49,13 +49,14 @@ const TXN_EVENT_SUBTYPE_VERSION_TO_VALIDATOR_MAPPING: Record<string, ValidatorIn
   [serializeSubtypeAndVersion(DydxIndexerSubtypes.UPDATE_PERPETUAL.toString(), 1)]: UpdatePerpetualValidator,
   [serializeSubtypeAndVersion(DydxIndexerSubtypes.UPDATE_CLOB_PAIR.toString(), 1)]: UpdateClobPairValidator,
   [serializeSubtypeAndVersion(DydxIndexerSubtypes.DELEVERAGING.toString(), 1)]: DeleveragingValidator,
-  [serializeSubtypeAndVersion(DydxIndexerSubtypes.OPEN_INTEREST_UPDATE.toString(), 1)]: OpenInterestUpdateValidator,
+  [serializeSubtypeAndVersion(DydxIndexerSubtypes.OPEN_INTEREST_UPDATE, 1)]: OpenInterestUpdateValidator,
```
Comment (Christopher-Li) [inline indexer/services/ender/src/lib/block-processor.ts:52]: why is this the only type that is not using `toString()`?

**Review by Christopher-Li — APPROVED** [review]

File: indexer/services/ender/src/lib/block-processor.ts (hunk 1 above)
Comment (vincentwschau) [reply indexer/services/ender/src/lib/block-processor.ts:52]: Bad revert. Will revert.

**Comment by vincentwschau** [comment]
@Mergifyio backport release/protocol/v5.x

**Comment by vincentwschau** [comment]
@Mergifyio backport release/indexer/v5.x

**Pull request merged**